src/devices/bus/isa/lbaenhancer.cpp

```diff
@@ -0,0 +1,142 @@
+       board with an empty ISA slot.
+
+    Supports 1 to 4 large hard disks. Bios boot remains possible.
+
+    Check date code of Ami BIOS (at bottom of boot screen):
+    40-0100-001139-00101111-111192-486ABC-F  (111192 will not support LBA)
+    50-0100-001292-00101111-072594-ABCDEF-F  (072594 will support LBA)
+
+    See: ftp://ami.com/archive/utility/LBA.TXT
+
+    Usage:
+      select type 47 for each LBA drive and set C,H,S (params) to '8', 
+      then repartition and reformat (Fdisk and Format). Data is lost.
+
+    Requirements:
+      Compatible OS (DOS >= 6.22) and LBA capable drive(s).
+***********************************************************************
+
+PCB Layout
+----------
+
+|-------------------------------------------|
+|                                           |
+|                                           |
+|                                           |
+|            EPROM     CD74LS245            |
+|                                  J3,J2,J1 |
+|                      PAL16V8              |
+|                                           |
+|                                           |
+|---|                                   |---|
+    |-----------------------------------|
+
+Notes:
+
+J3, J2, J1 set BIOS (start) address to C800, CC00, D000, D400, D800 or DC00
+
+- one CD74LS245 
+- one PALCE16V8H-25
+- one 32 K EPROM : 27C256 (STM) (mirrored after 16 K)
+
+*/
+
+#include "emu.h"
+#include "lbaenhancer.h"
+
+//**************************************************************************
+//  DEVICE DEFINITIONS
+//**************************************************************************
+
+DEFINE_DEVICE_TYPE(LBA_GEN, lba_gen_device, "lba_gen", "VS Systems LBA Enhancer BIOS 1995")
+
+//-------------------------------------------------
+//  ROM( lba_gen )
+//-------------------------------------------------
+
+ROM_START( lbabios )
+	ROM_REGION( 0xc8000, "lbabios", 0 )
+	ROM_LOAD( "lbaenhancer.bin", 0x0000, LBA_GEN_ROMSIZE, CRC(39d4566d) SHA1(d275193a870250f212dc29768d4e68fb43770e95) )
+ROM_END
+
+//-------------------------------------------------
+//  rom_region - device-specific ROM region
+//-------------------------------------------------
+
+const tiny_rom_entry *lba_gen_device::device_rom_region() const
+{
+	return ROM_NAME( lbabios );
+}
+
+//-------------------------------------------------
+//  device_add_mconfig - add device configuration
+//-------------------------------------------------
+MACHINE_CONFIG_START( lba_gen_device::device_add_mconfig )
+//	MCFG_DEVICE_ADD("lba_gen", LBA_GEN, 0)
+MACHINE_CONFIG_END
+
+
+//-------------------------------------------------
+//  dip switches  
+//-------------------------------------------------
+static INPUT_PORTS_START( lba_gen_dsw )
+
+	PORT_START("ROM_ADDRESS")
+	PORT_DIPNAME( 0xff, 0x00, "16 K ROM bank (disable shadowing; exclude in EMM386)")
+	PORT_DIPSETTING(    0, "C8000 - CBFFF (sometimes occupied by graphics or hard disk controller)" )
```

Comment:
These are too verbose and will be a problem in configuration menus.  Also, is this a real set of eight DIP switches or not?  If it represents the three configuration jumpers, it should be a CONFSETTING and be three bits wide, not eight.